File: code/leetcode.offer45.把数组排成最小的数/1.js
// ##剑指offer##levele
/**
 * @param {number[]} nums
 * @return {string}
 */
 var minNumber = function(nums) {
  return nums.sort((a, b) => {
      return `${a}${b}` > `${b}${a}` ? 1 : -1
  }).join('')
};

File: code/leetcode.offer53.0~n-1中缺失的数字/1.py
# ##剑指offer#二分法##levele
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        while l < r:
            mid = (l+r)//2
            if nums[r] == r:
                return nums[r]+1
            if nums[mid] == mid+1:
                r = mid
            else:
                l = mid+1
        return l+1 if nums[l] == l else l
